app/controllers/api/v1/projects_controller.rb

```diff
@@ -37,6 +37,31 @@ def create
         end
       end
 
+      def key
+        if params[:id].nil? || params[:contribution_key].nil?
+          respond_to do |format|
+            format.json { render json: { error: 'Neither Project ID nor Contribution Key can be empty.' }, status: :unprocessable_entity }
+          end
+        else
```

Comment:
Instead of an 'else' here, just have a return at the end of your 'if'

This way, you don't have to indent a level inward when you go through the other conditions.